hyprpm update - failed to install headers with error code 2

### Hyprland Version

<details> <summary>System/Version info</summary>

```
Hyprland, built from branch main at commit 1ed1ce9506e65f76fe5236194516f23d662bba0b  (internal: new shared_ptr and weak_ptr implementation (5883)).
Date: Sun May 5 18:16:00 2024
Tag: v0.40.0-9-g1ed1ce95, commits: 4615

flags: (if any)
```

</details>

### Bug or Regression?

Bug

### Description

```
[v] version returned: Hyprland, built from branch main at commit 1ed1ce9506e65f76fe5236194516f23d662bba0b  (internal: new shared_ptr and weak_ptr implementation (5883)).
Date: Sun May 5 18:16:00 2024
Tag: v0.40.0-9-g1ed1ce95, commits: 4615

flags: (if any)


[v] parsed commit 1ed1ce9506e65f76fe5236194516f23d662bba0b at branch main on Sun May 5 18:16:00 2024, commits 4615

! Cloning https://github.com/hyprwm/hyprland, this might take a moment.                                                                                                              
[v] will shallow since: dom apr 28 18:16:00 2024                                                                                                                                     

✔ cloned                                                                                                                                                                             
[v] git returned (co): Already on 'main'                                                                                                                                             
Your branch is up to date with 'origin/main'.

[v] git returned (rs): rm 'subprojects/tracy'                                                                                                                                        
HEAD is now at 1ed1ce9 internal: new shared_ptr and weak_ptr implementation (#5883)

✔ checked out to running ver                                                                                                                                                         
! configuring Hyprland                                                                                                                                                               
[v] setting PREFIX for cmake to /home/pbo/.local/share/hyprpm/headersRoot                                                                                                            

-- The C compiler identification is GNU 13.2.1
-- The CXX compiler identification is GNU 13.2.1
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /usr/bin/cc - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /usr/bin/c++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Gathering git info
-- Found Python3: /usr/bin/python3.12 (found version "3.12.3") found components: Interpreter
Building itab.c/itab.h...
-- Setting up wlroots
-- Found PkgConfig: /usr/bin/pkg-config (found version "2.1.1")
-- Found WaylandScanner at /usr/bin/wayland-scanner
-- Found wayland-protocols at //usr/share/wayland-protocols
-- Configuring Hyprland in Release with CMake
-- Checking deps...
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD - Success
-- Found Threads: TRUE
-- Found OpenGL: /usr/lib/libOpenGL.so  found components: GLES3
-- Checking for modules 'xkbcommon;uuid;wayland-server;wayland-client;wayland-cursor;wayland-protocols;cairo;pango;pangocairo;pixman-1;libdrm;libinput;hwdata;libseat;libdisplay-info;libliftoff;libudev;gbm;hyprwayland-scanner>=0.3.4;hyprlang>=0.3.2;hyprcursor>=0.1.7'
--   Found xkbcommon, version 1.7.0
--   Found uuid, version 2.40.0
--   Found wayland-server, version 1.22.91
--   Found wayland-client, version 1.22.91
--   Found wayland-cursor, version 1.22.91
--   Found wayland-protocols, version 1.36
--   Found cairo, version 1.18.0
--   Found pango, version 1.52.2
--   Found pangocairo, version 1.52.2
--   Found pixman-1, version 0.43.4
--   Found libdrm, version 2.4.120
--   Found libinput, version 1.25.0
--   Found hwdata, version 0.382
--   Found libseat, version 0.8.0
--   Found libdisplay-info, version 0.1.1
--   Found libliftoff, version 0.4.1
--   Found libudev, version 255
--   Found gbm, version 24.0.6-arch1.2
--   Found hyprwayland-scanner, version 0.3.4
--   Found hyprlang, version 0.5.1
--   Found hyprcursor, version 0.1.7
-- Looking for execinfo.h
-- Looking for execinfo.h - found
-- Configuration supports execinfo
-- Looking for backtrace in execinfo
-- Looking for backtrace in execinfo - not found
-- Looking for sys/timerfd.h
-- Looking for sys/timerfd.h - found
-- Checking for module 'epoll-shim'
--   Package 'epoll-shim', required by 'virtual:world', not found
-- XWAYLAND Enabled (NO_XWAYLAND not defined) checking deps...
-- Checking for modules 'xcb;xwayland;xcb-util;xcb-render;xcb-xfixes;xcb-icccm;xcb-composite;xcb-res;xcb-ewmh'
--   Found xcb, version 1.17.0
--   Found xwayland, version 21.1.99.1
--   Found xcb-util, version 0.4.1
--   Found xcb-render, version 1.17.0
--   Found xcb-xfixes, version 1.17.0
--   Found xcb-icccm, version 0.4.2
--   Found xcb-composite, version 1.17.0
--   Found xcb-res, version 1.17.0
--   Found xcb-ewmh, version 0.4.2
-- Checking for module 'xcb-errors'
--   Found xcb-errors, version 1.0.1
-- SYSTEMD support is requested (NO_SYSTEMD not defined)...
-- Setting precompiled headers
-- Setting link libraries
-- Checking for module 'tomlplusplus'
--   Found tomlplusplus, version 3.4.0
-- Configuring done (1.7s)
-- Generating done (0.0s)
-- Build files have been written to: /tmp/hyprpm/hyprland-pbo/build

[v] meson returned: The Meson build system                                                                                                                                           
Version: 1.4.0
Source dir: /tmp/hyprpm/hyprland-pbo/subprojects/wlroots-hyprland
Build dir: /tmp/hyprpm/hyprland-pbo/subprojects/wlroots-hyprland/build
Build type: native build
Project name: wlroots
Project version: 0.18.0-dev
C compiler for the host machine: cc (gcc 13.2.1 "cc (GCC) 13.2.1 20240417")
C linker for the host machine: cc ld.bfd 2.42.0
Host machine cpu family: x86_64
Host machine cpu: x86_64
Compiler for C supports arguments -Wundef: YES 
Compiler for C supports arguments -Wlogical-op: YES 
Compiler for C supports arguments -Wmissing-include-dirs: YES 
Compiler for C supports arguments -Wold-style-definition: YES 
Compiler for C supports arguments -Wpointer-arith: YES 
Compiler for C supports arguments -Winit-self: YES 
Compiler for C supports arguments -Wstrict-prototypes: YES 
Compiler for C supports arguments -Wimplicit-fallthrough=2: YES 
Compiler for C supports arguments -Wendif-labels: YES 
Compiler for C supports arguments -Wstrict-aliasing=2: YES 
Compiler for C supports arguments -Woverflow: YES 
Compiler for C supports arguments -Wmissing-prototypes: YES 
Compiler for C supports arguments -Walloca: YES 
Compiler for C supports arguments -Wno-missing-braces: YES 
Compiler for C supports arguments -Wno-missing-field-initializers: YES 
Compiler for C supports arguments -Wno-unused-parameter: YES 
Compiler for C supports arguments -fmacro-prefix-map=/prefix/to/hide=: YES 
Found pkg-config: YES (/usr/bin/pkg-config) 2.1.1
Run-time dependency wayland-server found: YES 1.22.91
Run-time dependency libdrm found: YES 2.4.120
Run-time dependency xkbcommon found: YES 1.7.0
Run-time dependency pixman-1 found: YES 0.43.4
Library m found: YES
Library rt found: YES
Run-time dependency wayland-protocols found: YES 1.36
Build-time dependency wayland-scanner found: YES 1.22.91
Program /usr/bin/wayland-scanner found: YES (/usr/bin/wayland-scanner)
Has header "linux/dma-buf.h" : YES 
Run-time dependency egl found: YES 1.5
Run-time dependency gbm found: YES 24.0.6-arch1.2
Run-time dependency glesv2 found: YES 3.2
Program ./embed.sh found: YES (/tmp/hyprpm/hyprland-pbo/subprojects/wlroots-hyprland/render/gles2/shaders/./embed.sh)
Dependency pixman-1 found: YES 0.43.4 (cached)
Dependency gbm found: YES 24.0.6-arch1.2 (cached)
Checking for function "gbm_bo_get_fd_for_plane" with dependency gbm: YES 
Run-time dependency libudev found: YES 255
Run-time dependency libseat found: YES 0.8.0
Build-time dependency hwdata found: YES 0.382
Run-time dependency libdisplay-info found: YES 0.1.1
Run-time dependency libliftoff found: YES 0.4.1
Run-time dependency libinput found: YES 1.25.0
Run-time dependency xcb found: YES 1.17.0
Run-time dependency xcb-dri3 found: YES 1.17.0
Run-time dependency xcb-present found: YES 1.17.0
Run-time dependency xcb-render found: YES 1.17.0
Run-time dependency xcb-renderutil found: YES 0.3.10
Run-time dependency xcb-shm found: YES 1.17.0
Run-time dependency xcb-xfixes found: YES 1.17.0
Run-time dependency xcb-xinput found: YES 1.17.0
Run-time dependency wayland-client found: YES 1.22.91
Run-time dependency xwayland found: YES 21.1.99.1
Dependency xcb found: YES 1.17.0 (cached)
Run-time dependency xcb-composite found: YES 1.17.0
Run-time dependency xcb-ewmh found: YES 0.4.2
Run-time dependency xcb-icccm found: YES 0.4.2
Dependency xcb-render found: YES 1.17.0 (cached)
Run-time dependency xcb-res found: YES 1.17.0
Dependency xcb-xfixes found: YES 1.17.0 (cached)
Run-time dependency xcb-errors found: YES 1.0.1
Checking for function "xcb_xfixes_set_client_disconnect_mode" with dependencies xcb, xcb-composite, xcb-ewmh, xcb-icccm, xcb-render, xcb-res, xcb-xfixes, xcb-errors: YES 
Configuring config.h using configuration
Configuring version.h using configuration
Configuring config.h using configuration
Message: Patches found. Applying...
Build targets in project: 136

wlroots 0.18.0-dev

    drm-backend     : YES
    x11-backend     : YES
    libinput-backend: YES
    xwayland        : YES
    gles2-renderer  : YES
    vulkan-renderer : NO
    gbm-allocator   : YES
    session         : YES
    xcb-errors      : YES
    egl             : YES
    libliftoff      : YES

  User defined options
    examples        : false
    renderers       : gles2

Found ninja-1.12.0 at /usr/bin/ninja
                                                 
✔ configured Hyprland                                                                                                                                                                
[v] installation will run: sed -i -e "s#PREFIX = /usr/local#PREFIX = /home/pbo/.local/share/hyprpm/headersRoot#" /tmp/hyprpm/hyprland-pbo/Makefile && cd /tmp/hyprpm/hyprland-pbo && make installheaders
 ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━╍━━━━━━━━━  4 / 5  Installing sources[v] installer returned: rm -fr /home/pbo/.local/share/hyprpm/headersRoot/include/hyprland              
mkdir -p /home/pbo/.local/share/hyprpm/headersRoot/include/hyprland
mkdir -p /home/pbo/.local/share/hyprpm/headersRoot/include/hyprland/protocols
mkdir -p /home/pbo/.local/share/hyprpm/headersRoot/include/hyprland/wlroots-hyprland
mkdir -p /home/pbo/.local/share/hyprpm/headersRoot/share/pkgconfig
find src -name '*.h*' -print0 | cpio --quiet -0dump /home/pbo/.local/share/hyprpm/headersRoot/include/hyprland
cd subprojects/wlroots-hyprland/include && find . -name '*.h*' -print0 | cpio --quiet -0dump /home/pbo/.local/share/hyprpm/headersRoot/include/hyprland/wlroots-hyprland && cd ../../..
cd subprojects/wlroots-hyprland/build/include && find . -name '*.h*' -print0 | cpio --quiet -0dump /home/pbo/.local/share/hyprpm/headersRoot/include/hyprland/wlroots-hyprland && cd ../../../..
cp ./protocols/*.h /home/pbo/.local/share/hyprpm/headersRoot/include/hyprland/protocols
cp ./protocols/*.hpp /home/pbo/.local/share/hyprpm/headersRoot/include/hyprland/protocols
cp ./build/hyprland.pc /home/pbo/.local/share/hyprpm/headersRoot/share/pkgconfig
if [ -d /usr/share/pkgconfig ]; then cp ./build/hyprland.pc /usr/share/pkgconfig 2>/dev/null || true; fi
chmod -R 755 /home/pbo/.local/share/hyprpm/headersRoot/include/hyprland
chmod 755 /home/pbo/.local/share/hyprpm/headersRoot/share/pkgconfig

✖ failed to install headers with error code 2                                                                                                                                        
 ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━  5 / 5  Failed                                                                                                                   

✖ Headers missing. Please run hyprpm update to fix those.
```

### How to reproduce

`hyprpm update -v`

### Crash reports, logs, images, videos

_No response_

I have the same issue here.
I have tryed manyally comiling and getting this:
```
❯ make all
g++ -shared -fPIC --no-gnu-unique main.cpp -o hyprwinwrap.so -g `pkg-config --cflags pixman-1 libdrm hyprland pangocairo libinput libudev wayland-server xkbcommon` -std=c++2b -O2
In file included from /usr/local/include/hyprland/src/includes.hpp:52,
                 from main.cpp:6:
/usr/include/wlr/types/wlr_layer_shell_v1.h:16:10: fatal error: wlr-layer-shell-unstable-v1-protocol.h: No such file or directory
   16 | #include "wlr-layer-shell-unstable-v1-protocol.h"
      |          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
compilation terminated.
make: *** [Makefile:2: all] Error 1
❯ cd ../hyprexpo
❯ make all
g++ -shared -fPIC --no-gnu-unique main.cpp overview.cpp -o hyprexpo.so -g `pkg-config --cflags pixman-1 libdrm hyprland pangocairo libinput libudev wayland-server xkbcommon` -std=c++2b -Wno-narrowing
In file included from /usr/local/include/hyprland/src/includes.hpp:52,
                 from /usr/local/include/hyprland/src/defines.hpp:1,
                 from /usr/local/include/hyprland/src/Compositor.hpp:7,
                 from main.cpp:5:
/usr/include/wlr/types/wlr_layer_shell_v1.h:16:10: fatal error: wlr-layer-shell-unstable-v1-protocol.h: No such file or directory
   16 | #include "wlr-layer-shell-unstable-v1-protocol.h"
      |          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
compilation terminated.
In file included from /usr/local/include/hyprland/src/includes.hpp:52,
                 from /usr/local/include/hyprland/src/defines.hpp:1,
                 from /usr/local/include/hyprland/src/plugins/PluginAPI.hpp:26,
                 from globals.hpp:3,
                 from overview.hpp:5,
                 from overview.cpp:1:
/usr/include/wlr/types/wlr_layer_shell_v1.h:16:10: fatal error: wlr-layer-shell-unstable-v1-protocol.h: No such file or directory
   16 | #include "wlr-layer-shell-unstable-v1-protocol.h"
      |          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
compilation terminated.
make: *** [Makefile:2: all] Error 1
```
It may be an issue installing the headers

> @LOSEARDES77 can you try using cmake to build the plugins?
> 
> ```
> $ cmake --build -S . -B build -G Ninja
> ```

That didn't work but doing:
```
cmake -S . -B build -G Ninja
cmake --build build
```
worked for hyprwinwrap but hyprexpo may work for other but i have a problem with with cmake and didn't work for me
output:
```
[1/2] Building CXX object CMakeFiles/hyprexpo.dir/overview.cpp.o
FAILED: CMakeFiles/hyprexpo.dir/overview.cpp.o 
/usr/bin/c++ -Dhyprexpo_EXPORTS -isystem /usr/include/hyprland/protocols -isystem /usr/include/hyprland/wlroots-hyprland -isystem /usr/include/libdrm -isystem /usr/include/pango-1.0 -isystem /usr/include/cairo -isystem /usr/include/glib-2.0 -isystem /usr/lib/glib-2.0/include -isystem /usr/include/harfbuzz -isystem /usr/include/freetype2 -isystem /usr/include/pixman-1 -isystem /usr/include/libpng16 -isystem /usr/include/libmount -isystem /usr/include/blkid -isystem /usr/include/sysprof-6 -isystem /usr/include/fribidi -std=gnu++23 -fPIC -pthread -MD -MT CMakeFiles/hyprexpo.dir/overview.cpp.o -MF CMakeFiles/hyprexpo.dir/overview.cpp.o.d -o CMakeFiles/hyprexpo.dir/overview.cpp.o -c /home/loseardes77/Repos/hyprland-plugins/hyprexpo/overview.cpp
/home/loseardes77/Repos/hyprland-plugins/hyprexpo/overview.cpp: In constructor ‘COverview::COverview(PHLWORKSPACE, bool)’:
/home/loseardes77/Repos/hyprland-plugins/hyprexpo/overview.cpp:166:82: warning: narrowing conversion of ‘(currentid % ((COverview*)this)->COverview::SIDE_LENGTH)’ from ‘int’ to ‘double’ [-Wnarrowing]
  166 |     pos.create((-((pMonitor->vecSize / (double)SIDE_LENGTH) * Vector2D{currentid % SIDE_LENGTH, currentid / SIDE_LENGTH}) * pMonitor->scale) * (pMonitor->vecSize / tileSize),
      |                                                                        ~~~~~~~~~~^~~~~~~~~~~~~
/home/loseardes77/Repos/hyprland-plugins/hyprexpo/overview.cpp:166:107: warning: narrowing conversion of ‘(currentid / ((COverview*)this)->COverview::SIDE_LENGTH)’ from ‘int’ to ‘double’ [-Wnarrowing]
  166 |     pos.create((-((pMonitor->vecSize / (double)SIDE_LENGTH) * Vector2D{currentid % SIDE_LENGTH, currentid / SIDE_LENGTH}) * pMonitor->scale) * (pMonitor->vecSize / tileSize),
      |                                                                                                 ~~~~~~~~~~^~~~~~~~~~~~~
/home/loseardes77/Repos/hyprland-plugins/hyprexpo/overview.cpp:208:73: error: no match for ‘operator=’ (operand types are ‘std::shared_ptr<std::function<void(void*, SCallbackInfo&, std::any)> >’ and ‘CSharedPointer<std::function<void(void*, SCallbackInfo&, std::any)> >’)
  208 |     mouseMoveHook = g_pHookSystem->hookDynamic("mouseMove", onCursorMove);
      |                                                                         ^
In file included from /usr/include/c++/13.2.1/memory:80,
                 from /usr/local/include/hyprland/src/plugins/HookSystem.hpp:5,
                 from /usr/local/include/hyprland/src/plugins/PluginAPI.hpp:24,
                 from /home/loseardes77/Repos/hyprland-plugins/hyprexpo/globals.hpp:3,
                 from /home/loseardes77/Repos/hyprland-plugins/hyprexpo/overview.hpp:5,
                 from /home/loseardes77/Repos/hyprland-plugins/hyprexpo/overview.cpp:1:
/usr/include/c++/13.2.1/bits/shared_ptr.h:418:9: note: candidate: ‘template<class _Yp> std::shared_ptr<_Tp>::_Assignable<const std::shared_ptr<_Yp>&> std::shared_ptr<_Tp>::operator=(const std::shared_ptr<_Yp>&) [with _Tp = std::function<void(void*, SCallbackInfo&, std::any)>]’
  418 |         operator=(const shared_ptr<_Yp>& __r) noexcept
      |         ^~~~~~~~
/usr/include/c++/13.2.1/bits/shared_ptr.h:418:9: note:   template argument deduction/substitution failed:
/home/loseardes77/Repos/hyprland-plugins/hyprexpo/overview.cpp:208:73: note:   ‘CSharedPointer<std::function<void(void*, SCallbackInfo&, std::any)> >’ is not derived from ‘const std::shared_ptr<_Tp>’
  208 |     mouseMoveHook = g_pHookSystem->hookDynamic("mouseMove", onCursorMove);
      |                                                                         ^
/usr/include/c++/13.2.1/bits/shared_ptr.h:429:9: note: candidate: ‘template<class _Yp> std::shared_ptr<_Tp>::_Assignable<std::auto_ptr<_Up> > std::shared_ptr<_Tp>::operator=(std::auto_ptr<_Up>&&) [with _Tp = std::function<void(void*, SCallbackInfo&, std::any)>]’
  429 |         operator=(auto_ptr<_Yp>&& __r)
      |         ^~~~~~~~
/usr/include/c++/13.2.1/bits/shared_ptr.h:429:9: note:   template argument deduction/substitution failed:
/home/loseardes77/Repos/hyprland-plugins/hyprexpo/overview.cpp:208:73: note:   ‘CSharedPointer<std::function<void(void*, SCallbackInfo&, std::any)> >’ is not derived from ‘std::auto_ptr<_Up>’
  208 |     mouseMoveHook = g_pHookSystem->hookDynamic("mouseMove", onCursorMove);
      |                                                                         ^
/usr/include/c++/13.2.1/bits/shared_ptr.h:446:9: note: candidate: ‘template<class _Yp> std::shared_ptr<_Tp>::_Assignable<std::shared_ptr<_Yp> > std::shared_ptr<_Tp>::operator=(std::shared_ptr<_Yp>&&) [with _Tp = std::function<void(void*, SCallbackInfo&, std::any)>]’
  446 |         operator=(shared_ptr<_Yp>&& __r) noexcept
      |         ^~~~~~~~
/usr/include/c++/13.2.1/bits/shared_ptr.h:446:9: note:   template argument deduction/substitution failed:
/home/loseardes77/Repos/hyprland-plugins/hyprexpo/overview.cpp:208:73: note:   ‘CSharedPointer<std::function<void(void*, SCallbackInfo&, std::any)> >’ is not derived from ‘std::shared_ptr<_Tp>’
  208 |     mouseMoveHook = g_pHookSystem->hookDynamic("mouseMove", onCursorMove);
      |                                                                         ^
/usr/include/c++/13.2.1/bits/shared_ptr.h:454:9: note: candidate: ‘template<class _Yp, class _Del> std::shared_ptr<_Tp>::_Assignable<std::unique_ptr<_Up, _Ep> > std::shared_ptr<_Tp>::operator=(std::unique_ptr<_Up, _Ep>&&) [with _Del = _Yp; _Tp = std::function<void(void*, SCallbackInfo&, std::any)>]’
  454 |         operator=(unique_ptr<_Yp, _Del>&& __r)
      |         ^~~~~~~~
/usr/include/c++/13.2.1/bits/shared_ptr.h:454:9: note:   template argument deduction/substitution failed:
/home/loseardes77/Repos/hyprland-plugins/hyprexpo/overview.cpp:208:73: note:   ‘CSharedPointer<std::function<void(void*, SCallbackInfo&, std::any)> >’ is not derived from ‘std::unique_ptr<_Tp, _Dp>’
  208 |     mouseMoveHook = g_pHookSystem->hookDynamic("mouseMove", onCursorMove);
      |                                                                         ^
/usr/include/c++/13.2.1/bits/shared_ptr.h:414:19: note: candidate: ‘std::shared_ptr<_Tp>& std::shared_ptr<_Tp>::operator=(const std::shared_ptr<_Tp>&) [with _Tp = std::function<void(void*, SCallbackInfo&, std::any)>]’
  414 |       shared_ptr& operator=(const shared_ptr&) noexcept = default;
      |                   ^~~~~~~~
/usr/include/c++/13.2.1/bits/shared_ptr.h:414:29: note:   no known conversion for argument 1 from ‘CSharedPointer<std::function<void(void*, SCallbackInfo&, std::any)> >’ to ‘const std::shared_ptr<std::function<void(void*, SCallbackInfo&, std::any)> >&’
  414 |       shared_ptr& operator=(const shared_ptr&) noexcept = default;
      |                             ^~~~~~~~~~~~~~~~~
/usr/include/c++/13.2.1/bits/shared_ptr.h:438:7: note: candidate: ‘std::shared_ptr<_Tp>& std::shared_ptr<_Tp>::operator=(std::shared_ptr<_Tp>&&) [with _Tp = std::function<void(void*, SCallbackInfo&, std::any)>]’
  438 |       operator=(shared_ptr&& __r) noexcept
      |       ^~~~~~~~
/usr/include/c++/13.2.1/bits/shared_ptr.h:438:30: note:   no known conversion for argument 1 from ‘CSharedPointer<std::function<void(void*, SCallbackInfo&, std::any)> >’ to ‘std::shared_ptr<std::function<void(void*, SCallbackInfo&, std::any)> >&&’
  438 |       operator=(shared_ptr&& __r) noexcept
      |                 ~~~~~~~~~~~~~^~~
/home/loseardes77/Repos/hyprland-plugins/hyprexpo/overview.cpp:209:73: error: no match for ‘operator=’ (operand types are ‘std::shared_ptr<std::function<void(void*, SCallbackInfo&, std::any)> >’ and ‘CSharedPointer<std::function<void(void*, SCallbackInfo&, std::any)> >’)
  209 |     touchMoveHook = g_pHookSystem->hookDynamic("touchMove", onCursorMove);
      |                                                                         ^
/usr/include/c++/13.2.1/bits/shared_ptr.h:418:9: note: candidate: ‘template<class _Yp> std::shared_ptr<_Tp>::_Assignable<const std::shared_ptr<_Yp>&> std::shared_ptr<_Tp>::operator=(const std::shared_ptr<_Yp>&) [with _Tp = std::function<void(void*, SCallbackInfo&, std::any)>]’
  418 |         operator=(const shared_ptr<_Yp>& __r) noexcept
      |         ^~~~~~~~
/usr/include/c++/13.2.1/bits/shared_ptr.h:418:9: note:   template argument deduction/substitution failed:
/home/loseardes77/Repos/hyprland-plugins/hyprexpo/overview.cpp:209:73: note:   ‘CSharedPointer<std::function<void(void*, SCallbackInfo&, std::any)> >’ is not derived from ‘const std::shared_ptr<_Tp>’
  209 |     touchMoveHook = g_pHookSystem->hookDynamic("touchMove", onCursorMove);
      |                                                                         ^
/usr/include/c++/13.2.1/bits/shared_ptr.h:429:9: note: candidate: ‘template<class _Yp> std::shared_ptr<_Tp>::_Assignable<std::auto_ptr<_Up> > std::shared_ptr<_Tp>::operator=(std::auto_ptr<_Up>&&) [with _Tp = std::function<void(void*, SCallbackInfo&, std::any)>]’
  429 |         operator=(auto_ptr<_Yp>&& __r)
      |         ^~~~~~~~
/usr/include/c++/13.2.1/bits/shared_ptr.h:429:9: note:   template argument deduction/substitution failed:
/home/loseardes77/Repos/hyprland-plugins/hyprexpo/overview.cpp:209:73: note:   ‘CSharedPointer<std::function<void(void*, SCallbackInfo&, std::any)> >’ is not derived from ‘std::auto_ptr<_Up>’
  209 |     touchMoveHook = g_pHookSystem->hookDynamic("touchMove", onCursorMove);
      |                                                                         ^
/usr/include/c++/13.2.1/bits/shared_ptr.h:446:9: note: candidate: ‘template<class _Yp> std::shared_ptr<_Tp>::_Assignable<std::shared_ptr<_Yp> > std::shared_ptr<_Tp>::operator=(std::shared_ptr<_Yp>&&) [with _Tp = std::function<void(void*, SCallbackInfo&, std::any)>]’
  446 |         operator=(shared_ptr<_Yp>&& __r) noexcept
      |         ^~~~~~~~
/usr/include/c++/13.2.1/bits/shared_ptr.h:446:9: note:   template argument deduction/substitution failed:
/home/loseardes77/Repos/hyprland-plugins/hyprexpo/overview.cpp:209:73: note:   ‘CSharedPointer<std::function<void(void*, SCallbackInfo&, std::any)> >’ is not derived from ‘std::shared_ptr<_Tp>’
  209 |     touchMoveHook = g_pHookSystem->hookDynamic("touchMove", onCursorMove);
      |                                                                         ^
/usr/include/c++/13.2.1/bits/shared_ptr.h:454:9: note: candidate: ‘template<class _Yp, class _Del> std::shared_ptr<_Tp>::_Assignable<std::unique_ptr<_Up, _Ep> > std::shared_ptr<_Tp>::operator=(std::unique_ptr<_Up, _Ep>&&) [with _Del = _Yp; _Tp = std::function<void(void*, SCallbackInfo&, std::any)>]’
  454 |         operator=(unique_ptr<_Yp, _Del>&& __r)
      |         ^~~~~~~~
/usr/include/c++/13.2.1/bits/shared_ptr.h:454:9: note:   template argument deduction/substitution failed:
/home/loseardes77/Repos/hyprland-plugins/hyprexpo/overview.cpp:209:73: note:   ‘CSharedPointer<std::function<void(void*, SCallbackInfo&, std::any)> >’ is not derived from ‘std::unique_ptr<_Tp, _Dp>’
  209 |     touchMoveHook = g_pHookSystem->hookDynamic("touchMove", onCursorMove);
      |                                                                         ^
/usr/include/c++/13.2.1/bits/shared_ptr.h:414:19: note: candidate: ‘std::shared_ptr<_Tp>& std::shared_ptr<_Tp>::operator=(const std::shared_ptr<_Tp>&) [with _Tp = std::function<void(void*, SCallbackInfo&, std::any)>]’
  414 |       shared_ptr& operator=(const shared_ptr&) noexcept = default;
      |                   ^~~~~~~~
/usr/include/c++/13.2.1/bits/shared_ptr.h:414:29: note:   no known conversion for argument 1 from ‘CSharedPointer<std::function<void(void*, SCallbackInfo&, std::any)> >’ to ‘const std::shared_ptr<std::function<void(void*, SCallbackInfo&, std::any)> >&’
  414 |       shared_ptr& operator=(const shared_ptr&) noexcept = default;
      |                             ^~~~~~~~~~~~~~~~~
/usr/include/c++/13.2.1/bits/shared_ptr.h:438:7: note: candidate: ‘std::shared_ptr<_Tp>& std::shared_ptr<_Tp>::operator=(std::shared_ptr<_Tp>&&) [with _Tp = std::function<void(void*, SCallbackInfo&, std::any)>]’
  438 |       operator=(shared_ptr&& __r) noexcept
      |       ^~~~~~~~
/usr/include/c++/13.2.1/bits/shared_ptr.h:438:30: note:   no known conversion for argument 1 from ‘CSharedPointer<std::function<void(void*, SCallbackInfo&, std::any)> >’ to ‘std::shared_ptr<std::function<void(void*, SCallbackInfo&, std::any)> >&&’
  438 |       operator=(shared_ptr&& __r) noexcept
      |                 ~~~~~~~~~~~~~^~~
/home/loseardes77/Repos/hyprland-plugins/hyprexpo/overview.cpp:211:79: error: no match for ‘operator=’ (operand types are ‘std::shared_ptr<std::function<void(void*, SCallbackInfo&, std::any)> >’ and ‘CSharedPointer<std::function<void(void*, SCallbackInfo&, std::any)> >’)
  211 |     mouseButtonHook = g_pHookSystem->hookDynamic("mouseButton", onCursorSelect);
      |                                                                               ^
/usr/include/c++/13.2.1/bits/shared_ptr.h:418:9: note: candidate: ‘template<class _Yp> std::shared_ptr<_Tp>::_Assignable<const std::shared_ptr<_Yp>&> std::shared_ptr<_Tp>::operator=(const std::shared_ptr<_Yp>&) [with _Tp = std::function<void(void*, SCallbackInfo&, std::any)>]’
  418 |         operator=(const shared_ptr<_Yp>& __r) noexcept
      |         ^~~~~~~~
/usr/include/c++/13.2.1/bits/shared_ptr.h:418:9: note:   template argument deduction/substitution failed:
/home/loseardes77/Repos/hyprland-plugins/hyprexpo/overview.cpp:211:79: note:   ‘CSharedPointer<std::function<void(void*, SCallbackInfo&, std::any)> >’ is not derived from ‘const std::shared_ptr<_Tp>’
  211 |     mouseButtonHook = g_pHookSystem->hookDynamic("mouseButton", onCursorSelect);
      |                                                                               ^
/usr/include/c++/13.2.1/bits/shared_ptr.h:429:9: note: candidate: ‘template<class _Yp> std::shared_ptr<_Tp>::_Assignable<std::auto_ptr<_Up> > std::shared_ptr<_Tp>::operator=(std::auto_ptr<_Up>&&) [with _Tp = std::function<void(void*, SCallbackInfo&, std::any)>]’
  429 |         operator=(auto_ptr<_Yp>&& __r)
      |         ^~~~~~~~
/usr/include/c++/13.2.1/bits/shared_ptr.h:429:9: note:   template argument deduction/substitution failed:
/home/loseardes77/Repos/hyprland-plugins/hyprexpo/overview.cpp:211:79: note:   ‘CSharedPointer<std::function<void(void*, SCallbackInfo&, std::any)> >’ is not derived from ‘std::auto_ptr<_Up>’
  211 |     mouseButtonHook = g_pHookSystem->hookDynamic("mouseButton", onCursorSelect);
      |                                                                               ^
/usr/include/c++/13.2.1/bits/shared_ptr.h:446:9: note: candidate: ‘template<class _Yp> std::shared_ptr<_Tp>::_Assignable<std::shared_ptr<_Yp> > std::shared_ptr<_Tp>::operator=(std::shared_ptr<_Yp>&&) [with _Tp = std::function<void(void*, SCallbackInfo&, std::any)>]’
  446 |         operator=(shared_ptr<_Yp>&& __r) noexcept
      |         ^~~~~~~~
/usr/include/c++/13.2.1/bits/shared_ptr.h:446:9: note:   template argument deduction/substitution failed:
/home/loseardes77/Repos/hyprland-plugins/hyprexpo/overview.cpp:211:79: note:   ‘CSharedPointer<std::function<void(void*, SCallbackInfo&, std::any)> >’ is not derived from ‘std::shared_ptr<_Tp>’
  211 |     mouseButtonHook = g_pHookSystem->hookDynamic("mouseButton", onCursorSelect);
      |                                                                               ^
/usr/include/c++/13.2.1/bits/shared_ptr.h:454:9: note: candidate: ‘template<class _Yp, class _Del> std::shared_ptr<_Tp>::_Assignable<std::unique_ptr<_Up, _Ep> > std::shared_ptr<_Tp>::operator=(std::unique_ptr<_Up, _Ep>&&) [with _Del = _Yp; _Tp = std::function<void(void*, SCallbackInfo&, std::any)>]’
  454 |         operator=(unique_ptr<_Yp, _Del>&& __r)
      |         ^~~~~~~~
/usr/include/c++/13.2.1/bits/shared_ptr.h:454:9: note:   template argument deduction/substitution failed:
/home/loseardes77/Repos/hyprland-plugins/hyprexpo/overview.cpp:211:79: note:   ‘CSharedPointer<std::function<void(void*, SCallbackInfo&, std::any)> >’ is not derived from ‘std::unique_ptr<_Tp, _Dp>’
  211 |     mouseButtonHook = g_pHookSystem->hookDynamic("mouseButton", onCursorSelect);
      |                                                                               ^
/usr/include/c++/13.2.1/bits/shared_ptr.h:414:19: note: candidate: ‘std::shared_ptr<_Tp>& std::shared_ptr<_Tp>::operator=(const std::shared_ptr<_Tp>&) [with _Tp = std::function<void(void*, SCallbackInfo&, std::any)>]’
  414 |       shared_ptr& operator=(const shared_ptr&) noexcept = default;
      |                   ^~~~~~~~
/usr/include/c++/13.2.1/bits/shared_ptr.h:414:29: note:   no known conversion for argument 1 from ‘CSharedPointer<std::function<void(void*, SCallbackInfo&, std::any)> >’ to ‘const std::shared_ptr<std::function<void(void*, SCallbackInfo&, std::any)> >&’
  414 |       shared_ptr& operator=(const shared_ptr&) noexcept = default;
      |                             ^~~~~~~~~~~~~~~~~
/usr/include/c++/13.2.1/bits/shared_ptr.h:438:7: note: candidate: ‘std::shared_ptr<_Tp>& std::shared_ptr<_Tp>::operator=(std::shared_ptr<_Tp>&&) [with _Tp = std::function<void(void*, SCallbackInfo&, std::any)>]’
  438 |       operator=(shared_ptr&& __r) noexcept
      |       ^~~~~~~~
/usr/include/c++/13.2.1/bits/shared_ptr.h:438:30: note:   no known conversion for argument 1 from ‘CSharedPointer<std::function<void(void*, SCallbackInfo&, std::any)> >’ to ‘std::shared_ptr<std::function<void(void*, SCallbackInfo&, std::any)> >&&’
  438 |       operator=(shared_ptr&& __r) noexcept
      |                 ~~~~~~~~~~~~~^~~
/home/loseardes77/Repos/hyprland-plugins/hyprexpo/overview.cpp:212:71: error: no match for ‘operator=’ (operand types are ‘std::shared_ptr<std::function<void(void*, SCallbackInfo&, std::any)> >’ and ‘CSharedPointer<std::function<void(void*, SCallbackInfo&, std::any)> >’)
  212 |     touchUpHook = g_pHookSystem->hookDynamic("touchUp", onCursorSelect);
      |                                                                       ^
/usr/include/c++/13.2.1/bits/shared_ptr.h:418:9: note: candidate: ‘template<class _Yp> std::shared_ptr<_Tp>::_Assignable<const std::shared_ptr<_Yp>&> std::shared_ptr<_Tp>::operator=(const std::shared_ptr<_Yp>&) [with _Tp = std::function<void(void*, SCallbackInfo&, std::any)>]’
  418 |         operator=(const shared_ptr<_Yp>& __r) noexcept
      |         ^~~~~~~~
/usr/include/c++/13.2.1/bits/shared_ptr.h:418:9: note:   template argument deduction/substitution failed:
/home/loseardes77/Repos/hyprland-plugins/hyprexpo/overview.cpp:212:71: note:   ‘CSharedPointer<std::function<void(void*, SCallbackInfo&, std::any)> >’ is not derived from ‘const std::shared_ptr<_Tp>’
  212 |     touchUpHook = g_pHookSystem->hookDynamic("touchUp", onCursorSelect);
      |                                                                       ^
/usr/include/c++/13.2.1/bits/shared_ptr.h:429:9: note: candidate: ‘template<class _Yp> std::shared_ptr<_Tp>::_Assignable<std::auto_ptr<_Up> > std::shared_ptr<_Tp>::operator=(std::auto_ptr<_Up>&&) [with _Tp = std::function<void(void*, SCallbackInfo&, std::any)>]’
  429 |         operator=(auto_ptr<_Yp>&& __r)
      |         ^~~~~~~~
/usr/include/c++/13.2.1/bits/shared_ptr.h:429:9: note:   template argument deduction/substitution failed:
/home/loseardes77/Repos/hyprland-plugins/hyprexpo/overview.cpp:212:71: note:   ‘CSharedPointer<std::function<void(void*, SCallbackInfo&, std::any)> >’ is not derived from ‘std::auto_ptr<_Up>’
  212 |     touchUpHook = g_pHookSystem->hookDynamic("touchUp", onCursorSelect);
      |                                                                       ^
/usr/include/c++/13.2.1/bits/shared_ptr.h:446:9: note: candidate: ‘template<class _Yp> std::shared_ptr<_Tp>::_Assignable<std::shared_ptr<_Yp> > std::shared_ptr<_Tp>::operator=(std::shared_ptr<_Yp>&&) [with _Tp = std::function<void(void*, SCallbackInfo&, std::any)>]’
  446 |         operator=(shared_ptr<_Yp>&& __r) noexcept
      |         ^~~~~~~~
/usr/include/c++/13.2.1/bits/shared_ptr.h:446:9: note:   template argument deduction/substitution failed:
/home/loseardes77/Repos/hyprland-plugins/hyprexpo/overview.cpp:212:71: note:   ‘CSharedPointer<std::function<void(void*, SCallbackInfo&, std::any)> >’ is not derived from ‘std::shared_ptr<_Tp>’
  212 |     touchUpHook = g_pHookSystem->hookDynamic("touchUp", onCursorSelect);
      |                                                                       ^
/usr/include/c++/13.2.1/bits/shared_ptr.h:454:9: note: candidate: ‘template<class _Yp, class _Del> std::shared_ptr<_Tp>::_Assignable<std::unique_ptr<_Up, _Ep> > std::shared_ptr<_Tp>::operator=(std::unique_ptr<_Up, _Ep>&&) [with _Del = _Yp; _Tp = std::function<void(void*, SCallbackInfo&, std::any)>]’
  454 |         operator=(unique_ptr<_Yp, _Del>&& __r)
      |         ^~~~~~~~
/usr/include/c++/13.2.1/bits/shared_ptr.h:454:9: note:   template argument deduction/substitution failed:
/home/loseardes77/Repos/hyprland-plugins/hyprexpo/overview.cpp:212:71: note:   ‘CSharedPointer<std::function<void(void*, SCallbackInfo&, std::any)> >’ is not derived from ‘std::unique_ptr<_Tp, _Dp>’
  212 |     touchUpHook = g_pHookSystem->hookDynamic("touchUp", onCursorSelect);
      |                                                                       ^
/usr/include/c++/13.2.1/bits/shared_ptr.h:414:19: note: candidate: ‘std::shared_ptr<_Tp>& std::shared_ptr<_Tp>::operator=(const std::shared_ptr<_Tp>&) [with _Tp = std::function<void(void*, SCallbackInfo&, std::any)>]’
  414 |       shared_ptr& operator=(const shared_ptr&) noexcept = default;
      |                   ^~~~~~~~
/usr/include/c++/13.2.1/bits/shared_ptr.h:414:29: note:   no known conversion for argument 1 from ‘CSharedPointer<std::function<void(void*, SCallbackInfo&, std::any)> >’ to ‘const std::shared_ptr<std::function<void(void*, SCallbackInfo&, std::any)> >&’
  414 |       shared_ptr& operator=(const shared_ptr&) noexcept = default;
      |                             ^~~~~~~~~~~~~~~~~
/usr/include/c++/13.2.1/bits/shared_ptr.h:438:7: note: candidate: ‘std::shared_ptr<_Tp>& std::shared_ptr<_Tp>::operator=(std::shared_ptr<_Tp>&&) [with _Tp = std::function<void(void*, SCallbackInfo&, std::any)>]’
  438 |       operator=(shared_ptr&& __r) noexcept
      |       ^~~~~~~~
/usr/include/c++/13.2.1/bits/shared_ptr.h:438:30: note:   no known conversion for argument 1 from ‘CSharedPointer<std::function<void(void*, SCallbackInfo&, std::any)> >’ to ‘std::shared_ptr<std::function<void(void*, SCallbackInfo&, std::any)> >&&’
  438 |       operator=(shared_ptr&& __r) noexcept
      |                 ~~~~~~~~~~~~~^~~
/home/loseardes77/Repos/hyprland-plugins/hyprexpo/overview.cpp: In member function ‘void COverview::redrawID(int, bool)’:
/home/loseardes77/Repos/hyprland-plugins/hyprexpo/overview.cpp:228:61: warning: narrowing conversion of ‘((COverview*)this)->COverview::GAP_WIDTH’ from ‘int’ to ‘double’ [-Wnarrowing]
  228 |     Vector2D tileRenderSize = (pMonitor->vecSize - Vector2D{GAP_WIDTH, GAP_WIDTH} * (SIDE_LENGTH - 1)) / SIDE_LENGTH;
      |                                                             ^~~~~~~~~
/home/loseardes77/Repos/hyprland-plugins/hyprexpo/overview.cpp:228:72: warning: narrowing conversion of ‘((COverview*)this)->COverview::GAP_WIDTH’ from ‘int’ to ‘double’ [-Wnarrowing]
  228 |     Vector2D tileRenderSize = (pMonitor->vecSize - Vector2D{GAP_WIDTH, GAP_WIDTH} * (SIDE_LENGTH - 1)) / SIDE_LENGTH;
      |                                                                        ^~~~~~~~~
/home/loseardes77/Repos/hyprland-plugins/hyprexpo/overview.cpp: In member function ‘void COverview::onDamageReported()’:
/home/loseardes77/Repos/hyprland-plugins/hyprexpo/overview.cpp:308:51: warning: narrowing conversion of ‘((COverview*)this)->COverview::GAP_WIDTH’ from ‘int’ to ‘double’ [-Wnarrowing]
  308 |     Vector2D    tileRenderSize = (SIZE - Vector2D{GAP_WIDTH, GAP_WIDTH} * (SIDE_LENGTH - 1)) / SIDE_LENGTH;
      |                                                   ^~~~~~~~~
/home/loseardes77/Repos/hyprland-plugins/hyprexpo/overview.cpp:308:62: warning: narrowing conversion of ‘((COverview*)this)->COverview::GAP_WIDTH’ from ‘int’ to ‘double’ [-Wnarrowing]
  308 |     Vector2D    tileRenderSize = (SIZE - Vector2D{GAP_WIDTH, GAP_WIDTH} * (SIDE_LENGTH - 1)) / SIDE_LENGTH;
      |                                                              ^~~~~~~~~
/home/loseardes77/Repos/hyprland-plugins/hyprexpo/overview.cpp: In member function ‘void COverview::close()’:
/home/loseardes77/Repos/hyprland-plugins/hyprexpo/overview.cpp:333:71: warning: narrowing conversion of ‘(((int)ID) % ((COverview*)this)->COverview::SIDE_LENGTH)’ from ‘int’ to ‘double’ [-Wnarrowing]
  333 |     pos  = (-((pMonitor->vecSize / (double)SIDE_LENGTH) * Vector2D{ID % SIDE_LENGTH, ID / SIDE_LENGTH}) * pMonitor->scale) * (pMonitor->vecSize / tileSize);
      |                                                                    ~~~^~~~~~~~~~~~~
/home/loseardes77/Repos/hyprland-plugins/hyprexpo/overview.cpp:333:89: warning: narrowing conversion of ‘(((int)ID) / ((COverview*)this)->COverview::SIDE_LENGTH)’ from ‘int’ to ‘double’ [-Wnarrowing]
  333 |     pos  = (-((pMonitor->vecSize / (double)SIDE_LENGTH) * Vector2D{ID % SIDE_LENGTH, ID / SIDE_LENGTH}) * pMonitor->scale) * (pMonitor->vecSize / tileSize);
      |                                                                                      ~~~^~~~~~~~~~~~~
/home/loseardes77/Repos/hyprland-plugins/hyprexpo/overview.cpp: In member function ‘void COverview::onSwipeUpdate(double)’:
/home/loseardes77/Repos/hyprland-plugins/hyprexpo/overview.cpp:432:74: warning: narrowing conversion of ‘(((COverview*)this)->COverview::openedID % ((COverview*)this)->COverview::SIDE_LENGTH)’ from ‘int’ to ‘double’ [-Wnarrowing]
  432 |         (-((pMonitor->vecSize / (double)SIDE_LENGTH) * Vector2D{openedID % SIDE_LENGTH, openedID / SIDE_LENGTH}) * pMonitor->scale) * (pMonitor->vecSize / tileSize);
      |                                                                 ~~~~~~~~~^~~~~~~~~~~~~
/home/loseardes77/Repos/hyprland-plugins/hyprexpo/overview.cpp:432:98: warning: narrowing conversion of ‘(((COverview*)this)->COverview::openedID / ((COverview*)this)->COverview::SIDE_LENGTH)’ from ‘int’ to ‘double’ [-Wnarrowing]
  432 |         (-((pMonitor->vecSize / (double)SIDE_LENGTH) * Vector2D{openedID % SIDE_LENGTH, openedID / SIDE_LENGTH}) * pMonitor->scale) * (pMonitor->vecSize / tileSize);
      |                                                                                         ~~~~~~~~~^~~~~~~~~~~~~
ninja: build stopped: subcommand failed.
```

I have this same issue when building hyprland so i have been doing it with meson and worked

> @matt1432 are you using hyprpm? Or the `plugins` option in Home Manager?

I am using the `plugins` option

Logs:
```console
@nix { "action": "setPhase", "phase": "unpackPhase" }
Running phase: unpackPhase
unpacking source archive /nix/store/d3q8pw123ph3wipgp3z24xmpvqp9j5ih-source
source root is source
@nix { "action": "setPhase", "phase": "patchPhase" }
Running phase: patchPhase
@nix { "action": "setPhase", "phase": "updateAutotoolsGnuConfigScriptsPhase" }
Running phase: updateAutotoolsGnuConfigScriptsPhase
@nix { "action": "setPhase", "phase": "configurePhase" }
Running phase: configurePhase
fixing cmake files...
cmake flags: -GNinja -DCMAKE_FIND_USE_SYSTEM_PACKAGE_REGISTRY=OFF -DCMAKE_FIND_USE_PACKAGE_REGISTRY=OFF -DCMAKE_EXPORT_NO_PACKAGE_REGISTRY=ON -DCMAKE_BUILD_TYPE=RelWithDebInfo -DBUILD_TESTING=OFF -DCMAKE_INSTALL_LOCALEDIR=/nix/store/m1wkvlk45nhxn210232v53302zg63jmz-hyprland-0.40.0+date=2024-05-07_6ccc221/share/locale -DCMAKE_INSTALL_LIBEXECDIR=/nix/store/m1wkvlk45nhxn210232v53302zg63jmz-hyprland-0.40.0+date=2024-05-07_6ccc221/libexec -DCMAKE_INSTALL_LIBDIR=/nix/store/m1wkvlk45nhxn210232v53302zg63jmz-hyprland-0.40.0+date=2024-05-07_6ccc221/lib -DCMAKE_INSTALL_DOCDIR=/nix/store/m1wkvlk45nhxn210232v53302zg63jmz-hyprland-0.40.0+date=2024-05-07_6ccc221/share/doc/Hyprland -DCMAKE_INSTALL_INFODIR=/nix/store/m1wkvlk45nhxn210232v53302zg63jmz-hyprland-0.40.0+date=2024-05-07_6ccc221/share/info -DCMAKE_INSTALL_MANDIR=/nix/store/nyw6zw7j320z5d3mb706s0nzvq481p1h-hyprland-0.40.0+date=2024-05-07_6ccc221-man/share/man -DCMAKE_INSTALL_OLDINCLUDEDIR=/nix/store/fif45d1nigjcilqcyq0a6hmdkjrgz64a-hyprland-0.40.0+date=2024-05-07_6ccc221-dev/include -DCMAKE_INSTALL_INCLUDEDIR=/nix/store/fif45d1nigjcilqcyq0a6hmdkjrgz64a-hyprland-0.40.0+date=2024-05-07_6ccc221-dev/include -DCMAKE_INSTALL_SBINDIR=/nix/store/m1wkvlk45nhxn210232v53302zg63jmz-hyprland-0.40.0+date=2024-05-07_6ccc221/sbin -DCMAKE_INSTALL_BINDIR=/nix/store/m1wkvlk45nhxn210232v53302zg63jmz-hyprland-0.40.0+date=2024-05-07_6ccc221/bin -DCMAKE_INSTALL_NAME_DIR=/nix/store/m1wkvlk45nhxn210232v53302zg63jmz-hyprland-0.40.0+date=2024-05-07_6ccc221/lib -DCMAKE_POLICY_DEFAULT_CMP0025=NEW -DCMAKE_OSX_SYSROOT= -DCMAKE_FIND_FRAMEWORK=LAST -DCMAKE_STRIP=/nix/store/4i0gb09kcanl3gbmqmm82iivbcjh2lmw-gcc-wrapper-13.2.0/bin/strip -DCMAKE_RANLIB=/nix/store/4i0gb09kcanl3gbmqmm82iivbcjh2lmw-gcc-wrapper-13.2.0/bin/ranlib -DCMAKE_AR=/nix/store/4i0gb09kcanl3gbmqmm82iivbcjh2lmw-gcc-wrapper-13.2.0/bin/ar -DCMAKE_C_COMPILER=gcc -DCMAKE_CXX_COMPILER=g++ -DCMAKE_INSTALL_PREFIX=/nix/store/m1wkvlk45nhxn210232v53302zg63jmz-hyprland-0.40.0+date=2024-05-07_6ccc221 -DNO_XWAYLAND:BOOL=FALSE -DLEGACY_RENDERER:BOOL=FALSE -DNO_SYSTEMD:BOOL=FALSE 
-- The C compiler identification is GNU 13.2.0
-- The CXX compiler identification is GNU 13.2.0
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /nix/store/4i0gb09kcanl3gbmqmm82iivbcjh2lmw-gcc-wrapper-13.2.0/bin/gcc - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /nix/store/4i0gb09kcanl3gbmqmm82iivbcjh2lmw-gcc-wrapper-13.2.0/bin/g++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Gathering git info
fatal: not a git repository (or any parent up to mount point /)
Stopping at filesystem boundary (GIT_DISCOVERY_ACROSS_FILESYSTEM not set).
fatal: not a git repository (or any parent up to mount point /)
Stopping at filesystem boundary (GIT_DISCOVERY_ACROSS_FILESYSTEM not set).
fatal: not a git repository (or any parent up to mount point /)
Stopping at filesystem boundary (GIT_DISCOVERY_ACROSS_FILESYSTEM not set).
fatal: not a git repository (or any parent up to mount point /)
Stopping at filesystem boundary (GIT_DISCOVERY_ACROSS_FILESYSTEM not set).
fatal: not a git repository (or any parent up to mount point /)
Stopping at filesystem boundary (GIT_DISCOVERY_ACROSS_FILESYSTEM not set).
fatal: not a git repository (or any parent up to mount point /)
Stopping at filesystem boundary (GIT_DISCOVERY_ACROSS_FILESYSTEM not set).
fatal: not a git repository (or any parent up to mount point /)
Stopping at filesystem boundary (GIT_DISCOVERY_ACROSS_FILESYSTEM not set).
[31mCMake Error at CMakeLists.txt:32 (add_subdirectory):
  The source directory

    /build/source/subprojects/udis86

  does not contain a CMakeLists.txt file.

[0m
-- Setting up wlroots
-- Found PkgConfig: /nix/store/vbk2n3094zyl9ywijks6a3d7s2i6wnfm-pkg-config-wrapper-0.29.2/bin/pkg-config (found version "0.29.2")
-- Found WaylandScanner at /nix/store/jsdqzw2g14h1wk6x93bl551drcd5i9ha-wayland-1.22.0-bin/bin/wayland-scanner
-- Found wayland-protocols at //nix/store/6v7nvm2lm6sy4ws9x4ksgyfjxxnpad9d-wayland-protocols-1.35/share/wayland-protocols
-- Configuring Hyprland in Release with CMake
-- Checking deps...
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD - Success
-- Found Threads: TRUE
-- Found OpenGL: /nix/store/n5plv6lv08bxshqq9ina34ckzklc7i81-libGL-1.7.0/lib/libOpenGL.so  found components: GLES3
-- Checking for modules 'xkbcommon;uuid;wayland-server;wayland-client;wayland-cursor;wayland-protocols;cairo;pango;pangocairo;pixman-1;libdrm;libinput;hwdata;libseat;libdisplay-info;libliftoff;libudev;gbm;hyprwayland-scanner>=0.3.4;hyprlang>=0.3.2;hyprcursor>=0.1.7'
--   Found xkbcommon, version 1.5.0
--   Found uuid, version 2.39.3
--   Found wayland-server, version 1.22.0
--   Found wayland-client, version 1.22.0
--   Found wayland-cursor, version 1.22.0
--   Found wayland-protocols, version 1.35
--   Found cairo, version 1.18.0
--   Found pango, version 1.52.2
--   Found pangocairo, version 1.52.2
--   Found pixman-1, version 0.43.4
--   Found libdrm, version 2.4.120
--   Found libinput, version 1.25.0
--   Found hwdata, version 0.381
--   Found libseat, version 0.8.0
--   Found libdisplay-info, version 0.1.1
--   Found libliftoff, version 0.4.1
--   Found libudev, version 255
--   Found gbm, version 24.0.5
--   Found hyprwayland-scanner, version 0.3.4
--   Found hyprlang, version 0.5.1
--   Found hyprcursor, version 0.1.7
-- Looking for execinfo.h
-- Looking for execinfo.h - found
-- Configuration supports execinfo
-- Looking for backtrace in execinfo
-- Looking for backtrace in execinfo - not found
-- Looking for sys/timerfd.h
-- Looking for sys/timerfd.h - found
-- Checking for module 'epoll-shim'
--   No package 'epoll-shim' found
-- XWAYLAND Enabled (NO_XWAYLAND not defined) checking deps...
-- Checking for modules 'xcb;xwayland;xcb-util;xcb-render;xcb-xfixes;xcb-icccm;xcb-composite;xcb-res;xcb-ewmh'
--   Found xcb, version 1.17.0
--   Found xwayland, version 23.2.6
--   Found xcb-util, version 0.4.1
--   Found xcb-render, version 1.17.0
--   Found xcb-xfixes, version 1.17.0
--   Found xcb-icccm, version 0.4.2
--   Found xcb-composite, version 1.17.0
--   Found xcb-res, version 1.17.0
--   Found xcb-ewmh, version 0.4.2
-- Checking for module 'xcb-errors'
--   Found xcb-errors, version 1.0.1
-- SYSTEMD support is requested (NO_SYSTEMD not defined)...
-- Setting precompiled headers
-- Setting link libraries
Could not open input file: No such file or directory
Could not open input file: No such file or directory
Could not open input file: No such file or directory
Could not open input file: No such file or directory
Couldn't load proto
-- Checking for module 'tomlplusplus'
--   Found tomlplusplus, version 3.4.0
-- Configuring incomplete, errors occurred!
```

--- a/hyprland.pc.in
+++ b/hyprland.pc.in
@@ -1,8 +1,7 @@
-prefix=@PREFIX@
-includedir=@INCLUDEDIR@
+prefix=@PREFIX@/@INCLUDEDIR@
 
 Name: Hyprland
 URL: https://github.com/hyprwm/Hyprland
 Description: Hyprland header files
 Version: @HYPRLAND_VERSION@
-Cflags: -I${includedir} -I${includedir}/hyprland/protocols -I${includedir}/hyprland
+Cflags: -I${prefix} -I${prefix}/hyprland/protocols -I${prefix}/hyprland -I${prefix}/hyprland/wlroots-hyprland

